src/ShopingCenter.java

```diff
@@ -0,0 +1,146 @@
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ShopingCenter {
+
+// *********** Static fields **********
+
+    private static int nextId = 0;
+
+// *********** End of Static fields ***********
+
+// *********** Fields ************
+
+    private String name;
+    private Address address;
+    private List<Shop> shopList;
+    private OpeningHours timeOpen;
+    private int id;
+
+// ********** End of fields ************
+
+// ********** Constructors for ShoppingCenter ***********
+
+    public ShopingCenter(String name, Address address, OpeningHours timeOpen) {
+        this.name = name;
+        this.address = address;
+        this.shopList = new LinkedList<Shop>();
+        this.timeOpen = timeOpen;
+        this.id = ShopingCenter.nextId++;
+    }
+
+    public ShopingCenter(String name, String address, int openTime, int closeTime) {
+        this(name, new Address(address), new OpeningHours(openTime, closeTime));
+    }
+
+// *********** End of Constructors ***********
+
+// ************ Getters and Setters **********
+
+    public int getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+// ********** End of G & S ***********
+
+// ********** Methods - CRUD for Shops **********
+    public void addShop(Shop newShop) {
+        shopList.add(newShop);
+    }
+
+    public Shop getShop(int shopId) {
+        for (Shop aShopList : shopList) {
+            if (aShopList.getId() == shopId) {
+                return aShopList;
+            }
+        }
+        System.out.println("Shop not found.");
+        return null;
+    }
+
+    public void updateShop(int shopId, Shop modifiedShop) {
+        deleteShop(shopId);
+        modifiedShop.setId(shopId);
+        addShop(modifiedShop);
+    }
+
+    public void deleteShop(int id) {
+        Shop toDelete = null;
+        for (Shop aShopList : shopList) {
```

Comment:
`aShopList`  should rather be a single `shop`

Reply:
Fixed. Shouldn't have used build-in Intelij fix.